### ISSUE TITLE
.env file for changing defaults of flags for CLI commands

### ISSUE THREAD

**Issue by shaunc** [devspace-cloud/devspace#977]
**Is your feature request related to a problem?**  

In one project, I build two images, one that depends on the 2nd. For this to work, I need to specify `--build-sequential` every time I build. Occasionally I forget and wait through a build only to have it fail.

**Which solution do you suggest?**  

In the configuration, there could be an `options` section, in which you could specify default options. E.g.:

```yaml
options:
  build:
    buildSequential: true
  deploy:
    buildSequential: true
```
There could be `all` or `global` sections as well, to specify given options for all commands for which it makes sense, or to specify specifically global options (respectively to `all` and `global` ... if `all` is supported, `global` need not be, of course).

**Which alternative solutions exist?**  

In fact I do have commands specified for build. When I switch back and forth between projects, I don't remember to use them.

**Additional context**  

/kind feature

**Comment by LukasGentele** [comment]
That is a pretty smart idea, I think. I would name the config section more specific, e.g. `defaultFlags`, `cliDefaults`, `commandDefaults` or something like that. In general, I think this is cool because users can customize the defaults for all relevant commands on a per-project basis. 

For teams, this would let config maintainers adjust the behavior of DevSpace to the needs of their developers without everyone having to remember to use the right flags and without having to define a whole bunch of custom commands which then are not easy to find in the documentation etc. => I am not a great fan of custom commands for just simple flags because it is hard for users to understand what happens and they cannot find anything about these commands in our docs. So, I think they should rather be limited to very specific workflows and tasks, e.g. migrate-database or something.

**Comment by shaunc** [comment]
Making them patchable by profile would also be nice -- e.g. to put in `--skip-push` for dev profile you use on local cluster.

**Comment by LukasGentele** [comment]
@shaunc Thanks again for suggesting this feature. It was not possible to add this to devspace.yaml because that would mess up the config loading process because flags are being processed before config loading and global flags such as `-s/-n` are already taking effect even before the config is loaded. That's why we ended up following the example of docker-compose and made DevSpace parse an .env file. Feel free to reopen if you see any issues with this approach or if anything is missing.

More information about the implementation here: https://devspace.sh/cli/docs/configuration/env-file

**Comment by shaunc** [comment]
Env file works for me. I've been submitting many fewer requests, because ... devspace is more or less working for me. :) Thanks for all the work!

**Comment by shaunc** [comment]
Oops ... I have in .env:
```
# DEVSPACE use build sequential
DEVSPACE_BUILD_FLAGS=--build-sequential
DEVSPACE_DEV_FLAGS=--build-sequential --skip-push
DEVSPACE_DEPLOY_FLAGS=--build-sequential
```
Then get:
```
% devspace build --skip-push 
[info]   Applying extra flags from environment: --build-sequential
[fatal]  unknown flag: --build-sequential
```

**Comment by FabianKramm** [comment]
@shaunc thanks for reporting! I'll fix this in the next release!

**Comment by FabianKramm** [comment]
@shaunc should be fixed with v4.12.2
